flinkprocessor/src/main/java/io/pravega/example/videoprocessor/FlinkFaceRecognizerJob.java

```diff
@@ -0,0 +1,293 @@
+
+            // Write chunks to Pravega encoded as JSON.
+            final FlinkPravegaWriter<ChunkedVideoFrame> flinkPravegaWriter = FlinkPravegaWriter.<ChunkedVideoFrame>builder()
+                    .withPravegaConfig(getConfig().getPravegaConfig())
+                    .forStream(getConfig().getOutputStreamConfig().getStream())
+                    .withSerializationSchema(new ChunkedVideoFrameSerializationSchema())
+                    .withEventRouter(frame -> String.format("%d", frame.camera))
+                    .withWriterMode(PravegaWriterMode.ATLEAST_ONCE)
+                    .build();
+            outChunkedVideoFrames
+                    .addSink(flinkPravegaWriter)
+                    .uid("output-sink")
+                    .name("output-sink");
+
+            log.info("Executing {} job", jobName);
+            env.execute(jobName);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    /**
+     * A map function that uses TensorFlow and pre-trained facenet model to locate the faces in video frame, and
+     * store the embeddings for the faces located.
+     * The TensorFlow Session cannot be serialized so it is declared transient and
+     * initialized in open().
+     */
+    static class FaceRecognizerMapFunctionVideoFrame extends RichMapFunction<VideoFrame, VideoFrame> {
+        final private static Logger log = LoggerFactory.getLogger(FlinkFaceRecognizerJob.FaceRecognizerMapFunctionVideoFrame.class);
+        private transient FaceRecognizer recognizer;
+
+        /**
+         * The first execution takes 6 minutes on a V100.
+         * We warmup in open() so that map() does not timeout.
+         */
+        @Override
+        public void open(Configuration parameters) throws Exception {
+            recognizer = new FaceRecognizer();
+            recognizer.warmup();
+        }
+
+        @Override
+        public void close() {
+            recognizer.close();
+        }
+
+        @Override
+        public VideoFrame map(VideoFrame frame) throws Exception {
+            log.info("map: BEGIN: camera={}, frameNumber={}", frame.camera, frame.frameNumber);
+            frame.recognizedBoxes = recognizer.locateFaces(frame.data);
+            for (BoundingBox faceLocation : frame.recognizedBoxes) {
+                Mat imageMat = imdecode(new Mat(frame.data), IMREAD_UNCHANGED);
+                byte[] currentFaceData = recognizer.cropFace(faceLocation, imageMat);
+                frame.embeddingValues.add(recognizer.embeddFace(currentFaceData));
+            }
+            log.info("map: END: camera={}, frameNumber={}", frame.camera, frame.frameNumber);
+            return frame;
+        }
+    }
+
+    /**
+     * A map function that uses TensorFlow and pre-trained facenet model to locate the faces in images within transactions, and
+     * store the embeddings for the faces located.
+     * The TensorFlow Session cannot be serialized so it is declared transient and
```

Comment:
This line does not apply to this class.

Reply:
In line 259, I am using a FaceRecognizer object which needs to be transient right? I need to do recognizer.warmup in the open() function